### PR TITLE
feat(laravel): search filter

### PR DESCRIPTION
**Pull request by soyuka**
| Q             | A
| ------------- | ---
| Branch?       | main
| License       | MIT
| Doc PR        |

### CODE REVIEW

File: src/Laravel/Eloquent/Extension/FilterQueryExtension.php [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+use ApiPlatform\State\ParameterNotFound;
+use Illuminate\Database\Eloquent\Builder;
+use Psr\Container\ContainerInterface;
+
+final readonly class FilterQueryExtension implements QueryExtensionInterface
+{
+    /**
+     * @param ContainerInterface<FilterInterface> $filterLocator
+     */
+    public function __construct(private ContainerInterface $filterLocator)
```
Comment (dunglas) [inline src/Laravel/Eloquent/Extension/FilterQueryExtension.php:27]:
```suggestion
    public function __construct(
    private readonly ContainerInterface $filterLocator,
)
```

File: src/Laravel/Eloquent/Serializer/SnakeCaseToCamelCaseNameConverter.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ */
+
+namespace ApiPlatform\Laravel\Eloquent\Serializer;
+
+use Symfony\Component\Serializer\Exception\UnexpectedPropertyException;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+/**
+ * CamelCase to Underscore name converter.
+ *
```
Comment (dunglas) [inline src/Laravel/Eloquent/Serializer/SnakeCaseToCamelCaseNameConverter.php:30]:
```suggestion
 *
 * @internal
```

File: src/Laravel/Eloquent/Serializer/SnakeCaseToCamelCaseNameConverter.php [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ * @author Aurélien Pillevesse <aurelienpillevesse@hotmail.fr>
+ */
+class SnakeCaseToCamelCaseNameConverter implements NameConverterInterface
```
Comment (dunglas) [inline src/Laravel/Eloquent/Serializer/SnakeCaseToCamelCaseNameConverter.php:34]:
```suggestion
final class SnakeCaseToCamelCaseNameConverter implements NameConverterInterface
```

File: src/Laravel/Eloquent/Serializer/SnakeCaseToCamelCaseNameConverter.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        private bool $lowerCamelCase = true,
+    ) {
+    }
+
+    /**
+     * @param class-string|null    $class
+     * @param string|null          $format
+     * @param array<string, mixed> $context
+     */
+    public function normalize(string $propertyName/* , ?string $class = null, ?string $format = null, array $context = [] */): string
```
Comment (dunglas) [inline src/Laravel/Eloquent/Serializer/SnakeCaseToCamelCaseNameConverter.php:56]: The BC layer can me removed.

File: src/State/Provider/ParameterProvider.php [hunk 5]
```diff
@@ -33,7 +34,7 @@ final class ParameterProvider implements ProviderInterface
 {
     use ParameterParserTrait;
 
-    public function __construct(private readonly ?ProviderInterface $decorated = null, private readonly ?ContainerInterface $locator = null)
+    public function __construct(private readonly PropertyNameCollectionFactoryInterface $propertyNameCollectionFactory, private readonly ?ProviderInterface $decorated = null, private readonly ?ContainerInterface $locator = null)
```
Comment (dunglas) [inline src/State/Provider/ParameterProvider.php:37]: BC break?

File: src/Laravel/Eloquent/Serializer/SnakeCaseToCamelCaseNameConverter.php [hunk 6]
```diff
@@ -0,0 +1,92 @@
+<?php
```
Comment (dunglas) [inline src/Laravel/Eloquent/Serializer/SnakeCaseToCamelCaseNameConverter.php:1]: Isn't this class a leftover of another patch?

File: src/Laravel/Eloquent/Extension/FilterQueryExtension.php (hunk 1 above)
Comment (soyuka) [reply src/Laravel/Eloquent/Extension/FilterQueryExtension.php:27]: class is readonly, fixed the constructor on several lines

**Pull request merged**